### PR TITLE
test: add `ecmul` and `ecpairing` integration tests

### LINKED ISSUE #377: Add integration tests for `ecAdd`, `ecMul` and `ecPairing` precompiles
Divided in #384, #385 and #387.

### PR DESCRIPTION
**Pull request by juanimedone**
Closes #377 (includes #385 and #387). Comes after #398. Work to be continued in #413.

## Changes
- Fix attempt to copy empty `return_data` to memory.
- Fix `ecadd`'s invalid points unit tests.
- Fix `ecadd`'s integration tests that return `PrecompileError`.
- Add integration tests for `ecmul`.
- Add integration tests for `ecpairing`.

### CODE REVIEW

**Review by lazcanoluca — APPROVED** [review]
Very thorough testing, nice

File: tests/evm.rs [hunk 1]
```diff
@@ -3465,6 +3452,702 @@ fn staticcall_on_precompile_ecmul_happy_path() {
...
+        Operation::StaticCall,
+    ];
+
+    let program = Program::from(caller_ops);
+    let caller_bytecode = Bytecode::from(program.to_bytecode());
+    let mut env = Env::default();
+    let db = Db::new().with_contract(caller_address, caller_bytecode);
+    env.tx.transact_to = TransactTo::Call(caller_address);
+
+    run_program_assert_num_result(env, db, BigUint::ZERO);
```
Comment (MegaRedHand) [inline tests/evm.rs:3640]: This `BigUint::ZERO` is due to the exit code being 0 on revert, right? We should add a comment mentioning that, or change it to `BigUint::from(constants::call_opcode::REVERT_RETURN_CODE)`.

**Review by MegaRedHand — APPROVED** [review]

**Comment by MegaRedHand** [comment]
Moved to draft to avoid accidental merging (this one's waiting for #398)

File: tests/evm.rs (hunk 1 above)
Comment (juanimedone) [reply tests/evm.rs:3640]: I originally thought so, but now while debugging that test I realized that the `ExecutionResult` is `Success` with `SuccessReason::Stop`, so I'm trying to pinpoint the issue.

**Review by lazcanoluca — APPROVED** [review]

**Review by TomasArrachea — APPROVED** [review]

**Pull request merged**